internal/services/devcenter/dev_center_network_connection_resource.go

```diff
@@ -0,0 +1,296 @@
+			}
+
+			id := networkconnections.NewNetworkConnectionID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := networkconnections.NetworkConnection{
+				Location: location.Normalize(model.Location),
+				Properties: &networkconnections.NetworkProperties{
+					DomainJoinType: networkconnections.DomainJoinType(model.DomainJoinType),
+					SubnetId:       pointer.To(model.SubnetId),
+				},
+				Tags: pointer.To(model.Tags),
+			}
+
+			if v := model.DomainName; v != "" {
+				parameters.Properties.DomainName = pointer.To(v)
+			}
+
+			if v := model.DomainPassword; v != "" {
+				parameters.Properties.DomainPassword = pointer.To(v)
+			}
+
+			if v := model.DomainUsername; v != "" {
+				parameters.Properties.DomainUsername = pointer.To(v)
+			}
+
+			if v := model.OrganizationUnit; v != "" {
+				parameters.Properties.OrganizationUnit = pointer.To(v)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r DevCenterNetworkConnectionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DevCenter.V20230401.NetworkConnections
+
+			id, err := networkconnections.ParseNetworkConnectionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := DevCenterNetworkConnectionResourceModel{
+				Name:              id.NetworkConnectionName,
+				ResourceGroupName: id.ResourceGroupName,
+				DomainPassword:    metadata.ResourceData.Get("domain_password").(string),
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					if v := props.DomainJoinType; v != "" {
+						state.DomainJoinType = string(v)
+					}
+
+					if v := props.SubnetId; v != nil {
+						state.SubnetId = pointer.From(v)
+					}
+
+					if v := props.DomainName; v != nil {
+						state.DomainName = pointer.From(v)
+					}
+
+					if v := props.DomainUsername; v != nil {
+						state.DomainUsername = pointer.From(v)
+					}
+
+					if v := props.OrganizationUnit; v != nil {
+						state.OrganizationUnit = pointer.From(v)
+					}
```

Comment:
`pointer.From` can handle the nil check for us so these can be simplified
```suggestion
	
					state.SubnetId = pointer.From(props.SubnetId)
					state.DomainName = pointer.From(props.DomainName)
					state.DomainUsername = pointer.From(props.DomainUsername)
					state.OrganizationUnit = pointer.From(props.OrganizationUnit)
```

Reply:
Updated